contracts/VetoGuard.sol

```diff
@@ -0,0 +1,237 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "./interfaces/IVetoGuard.sol";
+import "./interfaces/IVetoERC20Voting.sol";
+import "./interfaces/IGnosisSafe.sol";
+import "./TransactionHasher.sol";
+import "@gnosis.pm/zodiac/contracts/guard/BaseGuard.sol";
+import "@gnosis.pm/zodiac/contracts/factory/FactoryFriendly.sol";
+import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+
+/// @notice A guard contract that prevents transactions that have been vetoed from being executed on the Gnosis Safe
+contract VetoGuard is
+    TransactionHasher,
+    FactoryFriendly,
+    BaseGuard,
+    IVetoGuard
+{
+    uint256 public executionDelayBlocks;
+    IVetoERC20Voting public vetoERC20Voting;
+    IGnosisSafe public gnosisSafe;
+    mapping(bytes32 => uint256) transactionQueuedBlock;
+
+    constructor(
+        address _owner,
+        uint256 _executionDelayBlocks,
+        address _vetoERC20Voting,
+        address _gnosisSafe
+    ) {
+        bytes memory initializeParams = abi.encode(
+            _executionDelayBlocks,
+            _owner,
+            _vetoERC20Voting,
+            _gnosisSafe
+        );
+        setUp(initializeParams);
+    }
+
+    /// @notice Initialize function, will be triggered when a new proxy is deployed
+    /// @param initializeParams Parameters of initialization encoded
+    function setUp(bytes memory initializeParams) public override initializer {
```

Comment:
I am tad confused on the initialise Params here - why not just create an init function?

Reply:
That function is there for this contract to adhere to the FactoryFriendly interface:
https://github.com/gnosis/zodiac/blob/master/contracts/factory/FactoryFriendly.sol

Both of the Gnosis Zodiac guards - Meta & Scope, also inherit from this interface and implement that function. So it can be compatible with the factories.

Reply:
Perfect - would it be cool to just use this as the initialiser function? I hate constructors lol